packages/suite-desktop-core/src/modules/bioAuthModule.ts

```diff
@@ -3,189 +3,287 @@ import { systemPreferences } from 'electron';
-                    };
-                }
-
-                try {
-                    await systemPreferences.promptTouchID(params.message ?? PROMPT_REASON);
-
-                    return {
-                        success: true as const,
-                    };
-                } catch (error) {
-                    logger.info('bioAuth', `MAC: bioAuth validation failed: ${error}`);
-
-                    return {
-                        success: false as const,
-                        message: serializeError(error),
-                    };
-                }
-            },
-            { timeout: TIMEOUT },
-        ).catch(() => ({ success: false as const, message: 'timeout' }));
-    });
-
-    ipcMain.handle('bio-auth/is-bio-auth-available', ipcEvent => {
-        validateIpcMessage({ ipcEvent });
-
-        return scheduleAction(
-            async (): Promise<boolean> => {
-                try {
-                    const canPromptTouchID = await systemPreferences.canPromptTouchID();
-
-                    return canPromptTouchID;
-                } catch (error) {
-                    logger.info('bioAuth', `MAC: bioAuth isAvailable failed: ${error}`);
-
-                    return false;
-                }
-            },
-            { timeout: TIMEOUT },
-        ).catch(() => false);
-    });
+type ConstructorParams = {
+    logger: ILogger;
+    enabled: boolean;
 };
-
-const loadLinux = () => {
-    ipcMain.handle('bio-auth/validate-bio-auth', ipcEvent => {
-        validateIpcMessage({ ipcEvent });
-
-        return {
+abstract class BioAuth extends TypedEmitter<{
+    'validation-status-changed': (available: boolean) => void;
+}> {
+    constructor(params: ConstructorParams) {
+        super();
+        this.logger = params.logger;
+        this.enabled = params.enabled;
+    }
+    abstract isAvailable(): Promise<boolean>;
+    abstract validate(
+        message?: string,
+    ): Promise<{ success: true } | { success: false; message: string }>;
+    logger: ILogger;
+    enabled: boolean;
+
+    init(): Promise<void> {
+        return Promise.resolve();
+    }
+
+    private invalidationTimeouts: Record<string, NodeJS.Timeout> = {};
+
+    private validated: number = 0;
+
+    public setEnabled(enabled: boolean) {
+        this.enabled = enabled;
+    }
+
+    public isValidated() {
+        if (this.validated === 0) return false;
+
+        // has not been last validated longer than 24 hours ago
+        return Date.now() - this.validated < 1000 * 60 * 60 * 24;
+    }
+
+    private attachInvalidationTimeout(name: string, ms: number) {
+        if (!this.enabled) return;
+        this.clearInvalidationTimeout(name);
+        this.invalidationTimeouts[name] = setTimeout(() => {
+            this.logger.info('bioAuth', `invalidation timeout [${name}] triggered`);
+
+            this.validated = 0;
+            this.emit('validation-status-changed', this.isValidated());
```

Comment:
`isValidated()` always false because `validated=0`.
I understand that's how you mark the status as invalidated, by setting timestamp to 1.1.1970? Please add comment so it's clear that 0 means invalidated 🙏 
And if I understand correctly, `isValidated()` timestamp condition is just a precaution in case `invalidationTimeouts` failed to execute, is that right?
```suggestion
            this.emit('validation-status-changed', false);
```

Reply:
yes, and yes. 
I need to check 0 verbosely to avoid os time manipulations. I made some changes to this file. It still doesn't feel super clean and nice, thats right.